index.js

```diff
@@ -1,17 +1,81 @@
 class SortedList {
-  constructor() {}
+  constructor() {
+    this.items = [];
+    this.length = this.items.length;
+  }
 
-  add(item) {}
+  add(item) {
+    //Question to teacher: Why we cannot write the next two code lines in the same line? (this.items.push(item).sort((a,b) => a-b))
+    this.items.push(item);
+    this.items.sort((a,b) => a-b);
 
-  get(pos) {}
+    this.length ++;
+  }
 
-  max() {}
+  get(pos) {
+    if(pos >= this.items.length){
+      throw new Error('OutOfBounds');
+    } 
+    
+    else{
+      return this.items[pos];
+    }
+  }
 
-  min() {}
+  max() {
+    if(this.length === 0){
+      throw new Error('EmptySortedList');
+    } 
+    
+    else{
+      let max = -10000;
+      for (let i = 0; i < this.length; i++){
+        if(this.items[i] > max){
+          max = this.items[i];
+        }
+      }
 
-  sum() {}
+      return max;
+    }
+  }
 
-  avg() {}
+  min() {
+    if(this.length === 0){
+      throw new Error('EmptySortedList');
+    } 
+    
+    else {
+      let min = 10000;
+      for (let i = 0; i < this.length; i++){
+        if(this.items[i] < min){
+          min = this.items[i];
+        }
+      }
+
+      return min;
+    }
+  }
+  
+
+  sum() {
+    if(this.length === 0) {
+      return 0;
+    }
+
+    else {
+      return this.items.reduce((a, b) => a + b);
```

Comment:
Bien! pero le falta algo a este reduce!, el reduce necesita que le digas en que se tiene que transformar, al final, como quieres que te devuelva un numero, entonces seria asi
```suggestion
      return this.items.reduce((a, b) => a + b, 0);
```

Si quieras un array o un objeto seria [] o {} respectivamente.